File: old/0102_BinaryTreeLevelOrderTraversal.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def levelOrder(self, root):
        answer = []
        if root == None:
            return answer
        q = [root]

        while len(q) != 0:
            answer.append([ele.val for ele in q])
            tempq = []
            for ele in q:
                if ele.left:
                    tempq.append(ele.left)
                if ele.right:
                    tempq.append(ele.right)
            q = tempq
        return answer

File: top150/binary-tree-bfs/0102_Binary_Tree_Level_Order_Traversal.go
// Leetcode: https://leetcode.com/problems/binary-tree-level-order-traversal/
package binarytreebfs

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func levelOrder(root *TreeNode) [][]int {
	var res [][]int
	if root == nil {
		return res
	}

	queue := []*TreeNode{root}

	for len(queue) > 0 {
		n := len(queue)
		sub_res := []int{}
		for i := 0; i < n; i++ {
			node := queue[i]
			sub_res = append(sub_res, node.Val)
			if node.Left != nil {
				queue = append(queue, node.Left)
			}
			if node.Right != nil {
				queue = append(queue, node.Right)
			}
		}
		res = append(res, sub_res)
		queue = queue[n:]
	}

	return res
}
